app/src/main/java/de/tum/in/tumcampusapp/widgets/TimetableWidget.java

```diff
@@ -0,0 +1,140 @@
+        PendingIntent pendingConfigIntent = PendingIntent.getActivity(context, appWidgetId, configIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+        rv.setOnClickPendingIntent(R.id.timetable_widget_setting, pendingConfigIntent);
+
+        // Set up the calendar activity listeners
+        Intent calendarIntent = new Intent(context, CalendarActivity.class);
+        // todo fix day on which the calendar is opened
+        calendarIntent.putExtra(CalendarActivity.EVENT_TIME, 0);
+        PendingIntent pendingCalendarIntent = PendingIntent.getActivity(context, appWidgetId, calendarIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+        rv.setOnClickPendingIntent(R.id.timetable_widget_day, pendingCalendarIntent);
+
+        // Set up the intent that starts the TimetableWidgetService, which will
+        // provide the departure times for this station
+        Intent intent = new Intent(context, TimetableWidgetService.class);
+        intent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId);
+        intent.setData(Uri.parse(intent.toUri(Intent.URI_INTENT_SCHEME)));
+        rv.setRemoteAdapter(R.id.timetable_widget_listview, intent);
+
+        // The empty view is displayed when the collection has no items.
+        // It should be in the same layout used to instantiate the RemoteViews
+        // object above.
+        rv.setEmptyView(R.id.timetable_widget_listview, R.id.empty_list_item);
+
+        // Instruct the widget manager to update the widget
+        appWidgetManager.updateAppWidget(appWidgetId, rv);
+        appWidgetManager.notifyAppWidgetViewDataChanged(appWidgetId, R.id.timetable_widget_listview);
+    }
+
+    @Override
+    public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
+        updateAppWidgets(context, appWidgetManager, appWidgetIds);
+        setAlarm(context);
+        super.onUpdate(context, appWidgetManager, appWidgetIds);
+    }
+
+    @Override
+    public void onDeleted(Context context, int[] appWidgetIds) {
+        // When the user deletes the widget, delete the associated setting from the database.
+        super.onDeleted(context, appWidgetIds);
+    }
+
+    @Override
+    public void onEnabled(Context context) {
+        // Enter relevant functionality for when the first widget is created
+        setAlarm(context);
+    }
+
+    @Override
+    public void onDisabled(Context context) {
+        // Cancel alarm as the last widget has been removed
+        Intent intent = new Intent(context, TimetableWidget.class);
+        PendingIntent sender = PendingIntent.getBroadcast(context, 0, intent, 0);
+        AlarmManager am = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        am.cancel(sender);
+        alarmIsSet = false;
+        super.onDisabled(context);
+    }
+
+    @Override
+    public void onReceive(@NonNull Context context, @NonNull Intent intent) {
+        switch (intent.getAction()) {
+            case BROADCAST_ALARM_NAME:
```

Comment:
yes, but i assume, that with the configure activity there will be a second action, so this will fill up